runtime/compiler/ilgen/Walker.cpp

```diff
@@ -4153,6 +4153,20 @@ break
       return NULL;
       }
 
+   /**
+    * java/lang/invoke/MethodHandleImpl.profileBoolean() performs some internal profiling
+    * on the boolean parameter value before returning it.  Since the OpenJ9 JIT does not
+    * consume that profiling information the profiling overhead is not necessary.  Eliminate
+    * the call and simply replace it with a reference to the boolean parameter.
+    */
+   if (symbol->getRecognizedMethod() == TR::java_lang_invoke_MethodHandleImpl_profileBoolean)
+      {
+      pop();
+      TR::Node *resultNode = _stack->top();
```

Comment:
Is this a static method ? I ask, because there would be 3 values on the operand stack if it was an instance method and 2 values on the operand stack if it was a static method. The fact that we only see a single pop suggests that it may be a static method since we seem to be using the other value (boolean) that is at the top of the stack after the prior pop.

Reply:
Yes, both `profileBoolean` and `isCompileConstant` are static methods.

Reply:
The other question is around whether this transformation was better done in recognized call transformer, since that runs as part of IL gen opts and handles other transformations that are in the same general category. I'm probably okay letting this change in though I'd prefer someone taking a look across all these similar sounding places that are inline/fast pathing in somewhat similar ways : IL gen, inlining optimization, and recognized call transformer and do some sort of cleanup in the future. Unsafe fast path seems to be different in that it has some knowledge of calling context, and so I'm not including it in the prior list.

Reply:
I chose this spot because of the many other recognized method transformations happening in proximity to this.  I agree that centralizing where these kinds of transformations are occurring (and why) is the preferred long-term strategy.  I will create an issue to track that in the coming days.